bundles/org.openhab.core/src/main/java/org/openhab/core/library/types/QuantityTypeArithmeticGroupFunction.java

```diff
@@ -86,23 +97,27 @@ public State calculate(@Nullable Set<Item> items) {
                 return UnDefType.UNDEF;
             }
 
+            Unit<?> baseUnit = null;
             QuantityType<?> sum = null;
             int count = 0;
             for (Item item : items) {
-                if (isSameDimension(item)) {
-                    QuantityType itemState = item.getStateAs(QuantityType.class);
-                    if (itemState != null) {
-                        if (sum == null) {
-                            sum = itemState; // initialise the sum from the first item
-                            count++;
-                        } else {
-                            itemState = itemState.toInvertibleUnit(sum.getUnit());
-                            if (itemState != null) {
-                                sum = sum.add(itemState);
-                                count++;
-                            }
-                        }
-                    }
```

Comment:
As long as this check remains, none of the conversions to base unit are necessary. And you can use toUnit instead of toInvertibleUnit, which I would prefer.

If you want this to work for inverse units as well, you need to check for same dimension and inverse dimension. And that’s where it gets tricky. You may have to convert the item unit to its systemUnit (see discussion in other PR) and then the inverse to see if that is in the same dimension.

Reply:
I think we can simply continue if toInvertibleUnit returns null. (It does its own checks and falls through to toUnit anyway..)

Reply:
Yes, we can, but I don’t like it. If it calls this it kind of indicates this will also work with inverted units. And it does not because of the earlier check. Or we fix the checks and also allow inverse units, or we just call toUnit so it doesn’t create wrong expectations.

Reply:
Done